Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -998,20 +998,20 @@ private void HandleOrderEvent(OrderEvent fill)
 
                     try
                     {
-                        // to be called before updating the Portfolio
-                        NewOrderEvent?.Invoke(this, fill);
-
                         _algorithm.Portfolio.ProcessFill(fill);
                         _algorithm.TradeBuilder.ProcessFill(
                             fill,
                             securityConversionRate,
                             feeInAccountCurrency,
                             multiplier);
+
+                        // to be called after updating the Portfolio
+                        NewOrderEvent?.Invoke(this, fill);
```

Comment:
I'm wondering if there's any concern about silently breaking existing algorithms by changing the temporal behavior as to when this event is raised. I say this will silently break algorithms since it likely won't raise an error but rather would end in the algorithm tracking state incorrectly thereby silently corrupting the algorithm's state. _If_ this is a concern, then perhaps a _new_ event would be warranted, something like `NewOrderEventProcessed` (comments on event naming below in `<nit>`).

``` xml
<nit severity='low' type='naming-conventions' scope='events'>
```
unrelated to this PR -- `NewOrderEvent` is a **really really** _**really**_ bad event name. [Event names](https://docs.microsoft.com/en-us/dynamicsax-2012/developer/naming-conventions-delegates-and-event-handlers) should be verbs as either present tense (*ing, such as `OrderSubmitting`) and denote that the event was raised _right before_, so in this case, `OrderSubmitting` implies the event handler is invoked _before_ the order is actually submitted. Following completion of submitting the order, another event would be called `OrderSubmitted`. These past tense verbs imply that the described event has been completed.

`NewOrderEvent` is ambiguous. Is this a new order event was _initialized_ -- or was it _received_, or was it _processed_ or was it _something else_ ? It appears it is intended as not just a new order event, but actually a new _fill_ or _partial fill_ event. I think a better name for the event _as-is_ on `master` would be `OrderFillProcessing`  and this new event would be `OrderFillProcessed`

Making things even more ambiguous, the documentation for `NewOrderEvent` only states:
``` csharp
/// <summary>
/// Event fired when there is a new <see cref="QuantConnect.Orders.OrderEvent"/>
/// </summary>
/// <remarks>Will be called before the <see cref="SecurityPortfolioManager"/></remarks>
```
which doesn't even mention that the event is only raised for fill/partial fill `OrderEvent` instances.
``` xml
</nit>
```

Reply:
We (@Martin-Molinero and I) have discussed this. `NewOrderEvent` is a recent feature and it was only consumed by `OrderBasedInsightGenerator`, so we choose to refactor `OrderBasedInsightGenerator` instead of creating a new event (ref. #3571).

Reply:
Thanks for the reference -- not just for me, but for posterity 😄